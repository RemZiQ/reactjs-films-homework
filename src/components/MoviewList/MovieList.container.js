import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import {
  getInitFilms,
  getGenres,
  getFilms,
  getInitMovie,
  getCurrentMovie,
  getTrandingFilms,
  getTopRated,
  getComingSoonFilms,
} from '../../modules/module_Search/actions';
import MoviewList from './MovieList';

const mapStateToProps = state => ({ store: state });
const mapDispatchToProps = dispatch => ({
  fetchInitData: () => {
    dispatch(getInitFilms());
  },
  fetchInitMovie: () => {
    dispatch(getInitMovie());
  },
  fetchCurrentMovie: (id) => {
    dispatch(getCurrentMovie(id));
  },
  fetchData: (search) => {
    dispatch(getFilms(search));
  },
  fetchGengres: () => {
    dispatch(getGenres());
  },
  fetchTrandingData: (id) => {
    dispatch(getTrandingFilms(id));
  },
  fetchTopRatedData: () => {
    dispatch(getTopRated());
  },
  fetchComingSoonData: () => {
    dispatch(getComingSoonFilms());
  },
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(MoviewList));
