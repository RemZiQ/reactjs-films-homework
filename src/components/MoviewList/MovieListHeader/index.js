export { default } from './MovieListHeader.container';
