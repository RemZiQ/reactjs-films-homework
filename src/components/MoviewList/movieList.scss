.moviewList__container{
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  min-height: 1200px;
  padding: 20px 100px;
  background-color: #f1f1f1;
}