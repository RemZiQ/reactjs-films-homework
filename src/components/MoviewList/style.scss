.moviewList__container{
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  max-width: 100%;
  min-height: 1200px;
  padding: 20px;
  background-color: #f1f1f1;
}