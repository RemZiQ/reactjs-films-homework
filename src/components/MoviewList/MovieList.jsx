import React, { PureComponent, Component } from 'react';
import propTypes from 'prop-types';


import MovieListItem from '../MovieListItem';
import MovieListHeader from './MovieListHeader';
import './movieList.scss';


class MoviewList extends Component {
  constructor(props) {
    super(props);
    this.listOfMovieItems = '';
  }

  componentDidMount() {
    const {
      location,
      getInitFilms,
      getGenres,
      getFilms,
      getInitMovie,
      getCurrentMovie,
      getTrandingFilms,
      getTopRated,
      getComingSoonFilms,
      genres,
    } = this.props;

    if (location.pathname === '/') {
      if (!genres.length) {
        getGenres();
      }
      getInitFilms();
      getInitMovie();

      // needed for link`s share and get same result on the page
    } else if (location.pathname === '/search') {
      if (!genres.length) {
        getGenres();
      }
      const params = new URLSearchParams(location.search);
      const query = params.get('query');
      const id = params.get('id');
      switch (query) {
        case 'init793797112020979':
          getInitFilms();
          break;
        case 'Trending':
          getTrandingFilms();
          break;
        case 'Top Rated':
          getTopRated();
          break;
        case 'Coming soon':
          getComingSoonFilms();
          break;
        default:
          getFilms(query);
      }
      !!id ? getCurrentMovie(id) : getInitMovie();
    }
  }

  shouldComponentUpdate(nextProps) {
    if (this.props === nextProps) return false;
    return true;
  }

  render() {
    let listOfMovieItems = '';
    const { data } = this.props;
    if (data.results) {
      listOfMovieItems = data.results.map(item => (
        <MovieListItem
          key={item.id}
          imageUrl={item.poster_path}
          title={item.title}
          genres={item.genre_ids}
          mark={+(item.vote_average / 2)}
          id={item.id}
          overview={item.overview}
        />
      ));
    }

    return (
      <div>
        <MovieListHeader />
        <div className="moviewList__container">
          {listOfMovieItems}
        </div>
      </div>
    );
  }
}


MoviewList.propTypes = {
  data: propTypes.objectOf(propTypes.any).isRequired,
  genres: propTypes.arrayOf(propTypes.any).isRequired,
  location: propTypes.objectOf(propTypes.any),
  getInitFilms: propTypes.func.isRequired,
  getGenres: propTypes.func.isRequired,
  getFilms: propTypes.func.isRequired,
  getInitMovie: propTypes.func.isRequired,
  getCurrentMovie: propTypes.func.isRequired,
  getTrandingFilms: propTypes.func.isRequired,
  getTopRated: propTypes.func.isRequired,
  getComingSoonFilms: propTypes.func.isRequired,
};

MoviewList.defaultProps = {
  location: null,
};

export default MoviewList;
