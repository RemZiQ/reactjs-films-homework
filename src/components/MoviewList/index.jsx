import React from 'react';
import { connect } from 'react-redux';

import './style.scss';
import MovieListItem from '../MovieList__Item/index2';
// import state from '../../modules/store';


const MoviewList = ({ count, store }) => {
  console.log('________store.results', store.data.results);
  let listOfMovieItems = '';
  if(store.data.results){
    listOfMovieItems = store.data.results.map(item => {
      return <MovieListItem key={item.id} imageUrl={item.poster_path} 
      title={item.title} genres={item.genre_ids} mark={+(item.vote_average / 2)}
      id ={item.id} overview={item.overview}/>
    })
  }
  
return (
  <div className="moviewList__container">
  {listOfMovieItems}
  </div>
)
}

const mapStateToProps = (state) => {
  return {store: state}
}

export default connect(mapStateToProps)(MoviewList);
