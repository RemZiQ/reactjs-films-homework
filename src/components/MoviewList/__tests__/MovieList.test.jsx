import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import MoviewList from '../MovieList';


it('MoviewList renders well with out store', () => {
  const store = {
    data: { results: [] },
  };
  const renderer = new ShallowRenderer();
  const result = renderer.render(<MoviewList {...store} />);
  expect(result).toMatchSnapshot();
});

it('MoviewList renders well with store', () => {
  const store = {
    data: { results: [1, 2, 3] },
  };
  const renderer = new ShallowRenderer();
  const result = renderer.render(<MoviewList {...store} />);
  expect(result).toMatchSnapshot();
});
