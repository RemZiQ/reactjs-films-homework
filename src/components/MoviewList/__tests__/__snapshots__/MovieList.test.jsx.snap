// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviewList renders well with out store 1`] = `
<div>
  <Connect(MovieListHeader) />
  <div
    className="moviewList__container"
  />
</div>
`;

exports[`MoviewList renders well with store 1`] = `
<div>
  <Connect(MovieListHeader) />
  <div
    className="moviewList__container"
  >
    <Connect(MovieListItem)
      mark={NaN}
    />
    <Connect(MovieListItem)
      mark={NaN}
    />
    <Connect(MovieListItem)
      mark={NaN}
    />
  </div>
</div>
`;
